{
  "uid" : "43341908c4528f11",
  "name" : "test_insufficient_balance",
  "fullName" : "tests.brownie.token.tests.test_transferFrom#test_insufficient_balance",
  "historyId" : "d39c8b74877aa97c6c5dc1418b58c7a9",
  "time" : {
    "start" : 1642025673540,
    "stop" : 1642025673540,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "ValueError: connection already closed",
  "statusTrace" : "brownie_project = <Project 'TokenProject'>\naccounts = <brownie.network.account.Accounts object at 0x7f8c5f66dee0>\n\n    @pytest.fixture(scope=\"module\")\n    def token(brownie_project, accounts):\n>       return brownie_project.Token.deploy(\"Test Token\", \"TST\", 18, 1e21, {'from': accounts[0]})\n\ntests/brownie/token/tests/conftest.py:49: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/contract.py:600: in __call__\n    return tx[\"from\"].deploy(\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py:508: in deploy\n    receipt, exc = self._make_transaction(\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py:725: in _make_transaction\n    raise VirtualMachineError(e) from None\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = VirtualMachineError(ValueError({'code': -32000, 'message': 'connection already closed'}))\nexc = {'code': -32000, 'message': 'connection already closed'}\n\n    def __init__(self, exc: ValueError) -> None:\n        try:\n            exc = exc.args[0]\n        except Exception:\n            pass\n    \n        if isinstance(exc, dict) and \"message\" in exc:\n            if \"data\" not in exc:\n>               raise ValueError(exc[\"message\"]) from None\nE               ValueError: connection already closed\n\n../../../.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/exceptions.py:85: ValueError",
  "flaky" : false,
  "newFailed" : false,
  "newBroken" : false,
  "newPassed" : false,
  "retriesCount" : 0,
  "retriesStatusChange" : false,
  "beforeStages" : [ {
    "name" : "configure_neon_network",
    "time" : {
      "start" : 1642025653359,
      "stop" : 1642025653359,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "brownie_project",
    "time" : {
      "start" : 1642025654087,
      "stop" : 1642025669054,
      "duration" : 14967
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "accounts",
    "time" : {
      "start" : 1642025669821,
      "stop" : 1642025670091,
      "duration" : 270
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "setup_env",
    "time" : {
      "start" : 1642025669821,
      "stop" : 1642025669821,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "token",
    "time" : {
      "start" : 1642025672404,
      "stop" : 1642025672762,
      "duration" : 358
    },
    "status" : "broken",
    "statusMessage" : "ValueError: connection already closed\n",
    "statusTrace" : "  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/pluggy/_callers.py\", line 39, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/_pytest/fixtures.py\", line 1126, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/_pytest/fixtures.py\", line 932, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/neon-compatibility/neon-compatibility/tests/brownie/token/tests/conftest.py\", line 49, in token\n    return brownie_project.Token.deploy(\"Test Token\", \"TST\", 18, 1e21, {'from': accounts[0]})\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/contract.py\", line 600, in __call__\n    return tx[\"from\"].deploy(\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py\", line 508, in deploy\n    receipt, exc = self._make_transaction(\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/network/account.py\", line 725, in _make_transaction\n    raise VirtualMachineError(e) from None\n  File \"/home/runner/.local/share/virtualenvs/neon-compatibility-g882AfFF/lib/python3.8/site-packages/brownie/exceptions.py\", line 85, in __init__\n    raise ValueError(exc[\"message\"]) from None\n",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : true,
    "hasContent" : true
  } ],
  "afterStages" : [ {
    "name" : "accounts::0",
    "time" : {
      "start" : 1642025701832,
      "stop" : 1642025701832,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "feature",
    "value" : "Brownie"
  }, {
    "name" : "parentSuite",
    "value" : "tests.brownie.token.tests"
  }, {
    "name" : "suite",
    "value" : "test_transferFrom"
  }, {
    "name" : "host",
    "value" : "fv-az28-122"
  }, {
    "name" : "thread",
    "value" : "3766-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.brownie.token.tests.test_transferFrom"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 87,
        "skipped" : 0,
        "passed" : 0,
        "unknown" : 0,
        "total" : 87
      },
      "items" : [ {
        "uid" : "701265b5690584ed",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/472//#testresult/701265b5690584ed",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642018239296,
          "stop" : 1642018239296,
          "duration" : 0
        }
      }, {
        "uid" : "6bc9c45af3412c56",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/471//#testresult/6bc9c45af3412c56",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642015605204,
          "stop" : 1642015605204,
          "duration" : 0
        }
      }, {
        "uid" : "201cbd6ce9f8371e",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/470//#testresult/201cbd6ce9f8371e",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642014231323,
          "stop" : 1642014231323,
          "duration" : 0
        }
      }, {
        "uid" : "4f8f87ce7811d668",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/469//#testresult/4f8f87ce7811d668",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642012912805,
          "stop" : 1642012912805,
          "duration" : 0
        }
      }, {
        "uid" : "c956fd7d937d635a",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/468//#testresult/c956fd7d937d635a",
        "status" : "broken",
        "statusDetails" : "ValueError: connection already closed",
        "time" : {
          "start" : 1642011423644,
          "stop" : 1642011423644,
          "duration" : 0
        }
      }, {
        "uid" : "1368220855cdea56",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/466//#testresult/1368220855cdea56",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'execution reverted: Insufficient balance'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641998033248,
          "stop" : 1641998035766,
          "duration" : 2518
        }
      }, {
        "uid" : "bc7de918d7afd08c",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/465//#testresult/bc7de918d7afd08c",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'execution reverted: Insufficient balance'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641993794811,
          "stop" : 1641993797262,
          "duration" : 2451
        }
      }, {
        "uid" : "c225f3102686ee93",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/464//#testresult/c225f3102686ee93",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'execution reverted: Insufficient balance'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641991048388,
          "stop" : 1641991050795,
          "duration" : 2407
        }
      }, {
        "uid" : "e0a23c95e5dea167",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/463//#testresult/e0a23c95e5dea167",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'execution reverted: Insufficient balance'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641986393976,
          "stop" : 1641986395650,
          "duration" : 1674
        }
      }, {
        "uid" : "afee9a5f218a6d35",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/461//#testresult/afee9a5f218a6d35",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1895fe5e80>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641982479105,
          "stop" : 1641982479105,
          "duration" : 0
        }
      }, {
        "uid" : "acccf450abb9ab30",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/460//#testresult/acccf450abb9ab30",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f189460db20>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641980268144,
          "stop" : 1641980268144,
          "duration" : 0
        }
      }, {
        "uid" : "af13e1a3e1bd2fed",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/459//#testresult/af13e1a3e1bd2fed",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'execution reverted: Insufficient balance'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641975865716,
          "stop" : 1641975867877,
          "duration" : 2161
        }
      }, {
        "uid" : "e4bfb10bcf1b85bb",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/458//#testresult/e4bfb10bcf1b85bb",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f189a842400>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641974828225,
          "stop" : 1641974828225,
          "duration" : 0
        }
      }, {
        "uid" : "8c6f2b5b7508a5a1",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/457//#testresult/8c6f2b5b7508a5a1",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'HTTPConnectionPool(host='10.2.0.2', port=8899): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f1896042940>: Failed to establish a new connection: [Errno 111] Connection refused'))'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641974206809,
          "stop" : 1641974206809,
          "duration" : 0
        }
      }, {
        "uid" : "4a72fd600c633615",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/456//#testresult/4a72fd600c633615",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.ReadTimeout: HTTPConnectionPool(host='proxy.night.stand.neontest.xyz', port=80): Read timed out. (read timeout=30)",
        "time" : {
          "start" : 1641958012935,
          "stop" : 1641958012935,
          "duration" : 0
        }
      }, {
        "uid" : "191585f318aaf4bc",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/455//#testresult/191585f318aaf4bc",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.ReadTimeout: HTTPConnectionPool(host='proxy.night.stand.neontest.xyz', port=80): Read timed out. (read timeout=30)",
        "time" : {
          "start" : 1641952208987,
          "stop" : 1641952208987,
          "duration" : 0
        }
      }, {
        "uid" : "f3a92f60d5531e5c",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/454//#testresult/f3a92f60d5531e5c",
        "status" : "broken",
        "statusDetails" : "requests.exceptions.ReadTimeout: HTTPConnectionPool(host='proxy.night.stand.neontest.xyz', port=80): Read timed out. (read timeout=30)",
        "time" : {
          "start" : 1641950686656,
          "stop" : 1641950686656,
          "duration" : 0
        }
      }, {
        "uid" : "1b35128dab5e5ee5",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/453//#testresult/1b35128dab5e5ee5",
        "status" : "broken",
        "statusDetails" : "ValueError: ['Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]', 'Program log: program/src/storage_account.rs:76', 'Program log: program/src/account_data.rs:225', 'Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 21, trx 20', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 51427 of 500000 compute units', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]', 'Program log: program/src/storage_account.rs:76', 'Program log: program/src/account_data.rs:225', 'Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 21, trx 20', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 51427 of 500000 compute units', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]', 'Program log: program/src/storage_account.rs:76', 'Program log: program/src/account_data.rs:225', 'Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 21, trx 20', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 51427 of 500000 compute units', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]', 'Program log: program/src/storage_account.rs:76', 'Program log: program/src/account_data.rs:225', 'Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 21, trx 20', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 51427 of 500000 compute units', 'Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument']",
        "time" : {
          "start" : 1641947815890,
          "stop" : 1641947815890,
          "duration" : 0
        }
      }, {
        "uid" : "73b62e9fa0031b0e",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/452//#testresult/73b62e9fa0031b0e",
        "status" : "broken",
        "statusDetails" : "ValueError: {\"err\": {\"InstructionError\": [1, \"InvalidArgument\"]}, \"fee\": 10000, \"innerInstructions\": [], \"logMessages\": [\"Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io invoke [1]\", \"Program log: program/src/account_data.rs:225\", \"Program log: program/src/entrypoint.rs:1251 : Invalid Ethereum transaction nonce: acc 32, trx 31\", \"Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io consumed 20548 of 500000 compute units\", \"Program 53DfF883gyixYNXnM7s5xhdeyV8mVk9T4i2hGV9vG9io failed: invalid program argument\"], \"postBalances\": [5900140509360, 402670880, 2039280, 2039280, 2672640, 31967280, 2672640, 1, 0, 1, 1141440, 1130582400], \"postTokenBalances\": [{\"accountIndex\": 2, \"mint\": \"HPsV9Deocecw3GeZv1FkAPNCBRfuVyfw9MMwjwRe1xaU\", \"uiTokenAmount\": {\"amount\": \"939133358766988974\", \"decimals\": 9, \"uiAmount\": 939133358.7669889, \"uiAmountString\": \"939133358.766988974\"}}, {\"accountIndex\": 3, \"mint\": \"HPsV9Deocecw3GeZv1FkAPNCBRfuVyfw9MMwjwRe1xaU\", \"uiTokenAmount\": {\"amount\": \"0\", \"decimals\": 9, \"uiAmount\": null, \"uiAmountString\": \"0\"}}], \"preBalances\": [5900140519360, 402670880, 2039280, 2039280, 2672640, 31967280, 2672640, 1, 0, 1, 1141440, 1130582400], \"preTokenBalances\": [{\"accountIndex\": 2, \"mint\": \"HPsV9Deocecw3GeZv1FkAPNCBRfuVyfw9MMwjwRe1xaU\", \"uiTokenAmount\": {\"amount\": \"939133358766988974\", \"decimals\": 9, \"uiAmount\": 939133358.7669889, \"uiAmountString\": \"939133358.766988974\"}}, {\"accountIndex\": 3, \"mint\": \"HPsV9Deocecw3GeZv1FkAPNCBRfuVyfw9MMwjwRe1xaU\", \"uiTokenAmount\": {\"amount\": \"0\", \"decimals\": 9, \"uiAmount\": null, \"uiAmountString\": \"0\"}}], \"rewards\": [], \"status\": {\"Err\": {\"InstructionError\": [1, \"InvalidArgument\"]}}}",
        "time" : {
          "start" : 1641871044260,
          "stop" : 1641871098855,
          "duration" : 54595
        }
      }, {
        "uid" : "515b34a59f8a8ae7",
        "reportUrl" : "https://neonlabsorg.github.io/neon-compatibility/451//#testresult/515b34a59f8a8ae7",
        "status" : "broken",
        "statusDetails" : "ValueError: Gas estimation failed: 'execution reverted: Insufficient balance'. This transaction will likely revert. If you wish to broadcast, you must set the gas limit manually.",
        "time" : {
          "start" : 1641861490337,
          "stop" : 1641861492821,
          "duration" : 2484
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "43341908c4528f11.json",
  "parameterValues" : [ ]
}